packages/firebase_ml_vision/lib/src/barcode_detector.dart

```diff
@@ -4,18 +4,452 @@
+class VisionBarcodeFormat {
+  final int value;
+  const VisionBarcodeFormat._(this.value);
+
+  static const VisionBarcodeFormat All = const VisionBarcodeFormat._(0xFFFF);
+  static const VisionBarcodeFormat UnKnown = const VisionBarcodeFormat._(0);
+  static const VisionBarcodeFormat Code128 =
+      const VisionBarcodeFormat._(0x0001);
+  static const VisionBarcodeFormat Code39 = const VisionBarcodeFormat._(0x0002);
+  static const VisionBarcodeFormat Code93 = const VisionBarcodeFormat._(0x0004);
+  static const VisionBarcodeFormat CodaBar =
+      const VisionBarcodeFormat._(0x0008);
+  static const VisionBarcodeFormat DataMatrix =
+      const VisionBarcodeFormat._(0x0010);
+  static const VisionBarcodeFormat EAN13 = const VisionBarcodeFormat._(0x0020);
+  static const VisionBarcodeFormat EAN8 = const VisionBarcodeFormat._(0x0040);
+  static const VisionBarcodeFormat ITF = const VisionBarcodeFormat._(0x0080);
+  static const VisionBarcodeFormat QRCode = const VisionBarcodeFormat._(0x0100);
+  static const VisionBarcodeFormat UPCA = const VisionBarcodeFormat._(0x0200);
+  static const VisionBarcodeFormat UPCE = const VisionBarcodeFormat._(0x0400);
+  static const VisionBarcodeFormat PDF417 = const VisionBarcodeFormat._(0x0800);
+  static const VisionBarcodeFormat Aztec = const VisionBarcodeFormat._(0x1000);
+}
+
+enum VisionBarcodeValueType {
+  /// Unknown Barcode value types.
+  Unknown,
+
+  /// Barcode value type for contact info.
+  ContactInfo,
+
+  /// Barcode value type for email addresses.
+  Email,
+
+  /// Barcode value type for ISBNs.
+  ISBN,
+
+  /// Barcode value type for phone numbers.
+  Phone,
+
+  /// Barcode value type for product codes.
+  Product,
+
+  /// Barcode value type for SMS details.
+  SMS,
+
+  /// Barcode value type for plain text.
+  Text,
+
+  /// Barcode value type for URLs/bookmarks.
+  URL,
+
+  /// Barcode value type for Wi-Fi access point details.
+  WiFi,
+
+  /// Barcode value type for geographic coordinates.
+  GeographicCoordinates,
+
+  /// Barcode value type for calendar events.
+  CalendarEvent,
+
+  /// Barcode value type for driver's license data.
+  DriversLicense,
+}
+
+// ios
+//   https://firebase.google.com/docs/reference/ios/firebasemlvision/api/reference/Classes/FIRVisionBarcodeEmail
+// android
+//   https://firebase.google.com/docs/reference/android/com/google/firebase/ml/vision/barcode/FirebaseVisionBarcode.Email
+class VisionBarcodeEmail {
+  VisionBarcodeEmail._(Map<dynamic, dynamic> data)
+      : type = VisionBarcodeEmailType.values.elementAt(data['type']),
+        address = data['address'] != null ? data['address'] : null,
+        body = data['body'] != null ? data['body'] : null,
+        subject = data['subject'] != null ? data['subject'] : null;
+
+  final String address;
+  final String body;
+  final String subject;
+  final VisionBarcodeEmailType type;
+}
+
+enum VisionBarcodeEmailType {
```

Comment:
Could you add a little documentation for the enums in the class as well. The enum is straight forward so you could just simply use `/// The type of email for [VisionBarcodeEmail]`